File: strategyDemo/src/main/ruby/strategy.rb
# build map of strategies
strategies = {
  :noop =>   lambda {},
  :stdout => lambda { |msg| puts msg },
  :sterr =>  lambda { |msg| $stderr.puts msg },
  :time =>   lambda { |msg| puts "#{Time.now} #{msg}" }
  }

# demo call
strategies[:time].call "test"

File: strategyDemo/src/main/c/strategy.c
#include <stdio.h>
#include <time.h>

typedef void (*strategy)(char *);

void noop(char* msg) {
}

void standart_output(char* msg) {
  printf("%s\n", msg);
}

void standart_error(char* msg) {
  fprintf(stderr, "%s\n", msg);
}

void with_time(char* msg) {
  time_t t = time(NULL);
  printf("%d: %s\n", (int)t, msg);
}

int main(void) {
  char* msg = "Hello, World!";
  strategy strategies[4] = {
     &noop,
     &standart_output,
     &standart_error,
     &with_time
  };
  int i;
  for (i=0; i<4; i++) {
     strategies[i](msg);
  }
}
